orchagent/aclorch.cpp

```diff
@@ -493,17 +510,110 @@ bool AclRuleL3::validateAddAction(string attr_name, string _attr_value)
     {
         value.aclaction.parameter.s32 = SAI_PACKET_ACTION_DROP;
     }
+    else if (attr_value.find(PACKET_ACTION_REDIRECT) != string::npos)
+    {
+        // resize attr_value to remove argument, _attr_value still has the argument
+        attr_value.resize(string(PACKET_ACTION_REDIRECT).length());
+
+        sai_object_id_t param_id = getRedirectObjectId(_attr_value);
+        if (param_id == SAI_NULL_OBJECT_ID)
+        {
+            return false;
+        }
+        value.aclaction.parameter.oid = param_id;
+    }
     else
     {
         return false;
     }
     value.aclaction.enable = true;
 
-    m_actions[aclActionLookup[attr_name]] = value;
+    m_actions[aclL3ActionLookup[attr_value]] = value;
 
     return true;
 }
 
+// This method should return sai attribute id of the redirect destination
+sai_object_id_t AclRuleL3::getRedirectObjectId(const string& redirect_value)
+{
+    // check that we have a colon after redirect rule
+    size_t colon_pos = string(PACKET_ACTION_REDIRECT).length();
+    if (redirect_value[colon_pos] != ':')
+    {
+        SWSS_LOG_ERROR("Redirect action rule must have ':' after REDIRECT");
+        return SAI_NULL_OBJECT_ID;
+    }
+
+    if (colon_pos + 1 == redirect_value.length())
+    {
+        SWSS_LOG_ERROR("Redirect action rule must have a target after 'REDIRECT:' action");
+        return SAI_NULL_OBJECT_ID;
+    }
+
+    string target = redirect_value.substr(colon_pos + 1);
```

Comment:
1. Because substr is a standard method of std::string.
2. We're 100% sure here that we have PACKET_ACTION_REDIRECT so assert is not required.
3. We're 100% sure here that the redirect action is not empty.
Also assert will crash or do nothing (depends on flags). When I check each case individually I output log messages which show users what was the error in their case. I think it's valuable because it allows an user to fix their configuration and try again